relay-server/src/services/processor.rs

```diff
@@ -2193,12 +2193,58 @@ impl EnvelopeProcessorService {
             sent_at,
         } = message;
 
+        struct Buckets {
+            data: SmallVec<[(ProjectKey, MetricData); 1]>,
+            feature_weights: FeatureWeights,
+        }
+
+        impl<'de> serde::Deserialize<'de> for Buckets {
```

Comment:
Could you add a code comment here that explains why we have a custom deserializer? Essentially what you wrote in the PR description.